kanidmd/src/lib/constants/ldap.rs

```diff
@@ -0,0 +1,192 @@
+    ///  The user cannot change the password. Note: You cannot assign the permission settings of PasswordCantChange by directly modifying the UserAccountControl attribute. For more information and a code example that shows how to prevent a user from changing the password, see User Cannot Change Password.
+    PasswordCantChange,
+    ///  The user can send an encrypted password.
+    EncryptedTextPasswordAllowed,
+    ///  This is an account for users whose primary account is in another AD DOMAIN. This account provides user access to this AD DOMAIN, but not to any AD DOMAIN that trusts this AD DOMAIN. Also known as a local user account.
+    TemplDuplicateAccount,
+    /// This is a default account type that represents a typical user.
+    NormalAccount,
+    ///  This is a permit to trust account for a system AD DOMAIN that trusts other AD DOMAIN.
+    InterDomainTrustAccount,
+    ///  This is a computer account for a computer that is a member of this AD DOMAIN.
+    WorkstationTrustAccount,
+    ///  This is a computer account for a system backup Domain Controller that is a member of this AD DOMAIN.
+    ServerTrustAccount,
+    ///  The password for this account will never expire.
+    DontExpirePassword,
+    ///  This is an MNS logon account.
+    MnsLogonAccount,
+    ///  The user must log on using a Smart Card.
+    SmartcardRequired,
+    ///  The service account (user or computer account), under which a service runs, is trusted for Kerberos delegation. Any such service can impersonate a client requesting the service.
+    TrustedForDelegation,
+    ///  The security context of the user will NOT be delegated to a service even if the service account is set as trusted for Kerberos delegation.
+    NotDelgated,
+    ///  Restrict this UserPrincipalName to use only Data Encryption Standard (DES) encryption types for keys.
+    UseDesKeyOnly,
+    ///  This account does not require Kerberos Pre-Authentication for logon.
+    DontRequirePreAuth,
+    ///  The user password has expired. This flag is created by the system using data from the Pwd-Last-Set attribute and the AD DOMAIN policy.
+    ErrorPasswordExpired,
+    ///  The account is enabled for delegation. This is a security-sensitive setting; accounts with this option enabled SHOULD be strictly controlled. This setting enables a service running under the account to assume a client identity and authenticate as that user to other remote servers on the network.
+    TrustedToAuthnForDelegation,
+    ///  (Windows Server 2008/Windows Server 2008 R2) The account is a Read-Only Domain Controller (RODC). This is a security-sensitive setting. Removing this setting from an RODC compromises security on that server.
+    PartialSecretsAccount,
+    ///  Restrict this UserPrincipalName to use only Advanced Encryption Standard (AES) encryption types for keys. This bit is ignored by Windows Client and Windows Servers.
+    UserUseAesKeys,
+}
+
+impl From<LdapUacFlag> for u32 {
```

Comment:
I think you could do LdapUacFlag as repr=u32 instead here?

Reply:
I really don't know enough rust to know what you mean by that, and google's not helping sorry :)

Reply:
.. fixed :D